projects/plugins/jetpack/sal/class.json-api-post-base.php

```diff
@@ -1,74 +1,208 @@
+require_once __DIR__ . '/class.json-api-date.php';
 require_once ABSPATH . 'wp-admin/includes/post.php';
 require_once ABSPATH . 'wp-includes/post.php';
 
+/**
+ * Base class for SAL_Post.
+ */
 abstract class SAL_Post {
+
+	/**
+	 * A WP_Post instance.
+	 *
+	 * @var WP_Post
+	 */
 	public $post;
+
+	/**
+	 * The post request context (for example 'edit' or 'display')
+	 *
+	 * @var string
+	 */
 	public $context;
+
+	/**
+	 * A Jetpack_Site instance.
+	 *
+	 * @var Jetpack_Site
+	 */
 	public $site;
 
-	function __construct( $site, $post, $context ) {
-		$this->post = $post;
+	/**
+	 * Constructor function
+	 *
+	 * @param Jetpack_Site $site A Jetpack_Site instance.
+	 * @param WP_Post      $post A WP_Post instance.
+	 * @param string       $context The post request context (for example 'edit' or 'display').
+	 */
+	public function __construct( $site, $post, $context ) {
+		$this->post    = $post;
 		$this->context = $context;
-		$this->site = $site;
+		$this->site    = $site;
 	}
 
+	/**
+	 * Setting this WP_Post instance's key value
+	 *
+	 * @param string $key The post key to set.
+	 * @param string $value The value to set the post key to (for example filter, ID, post_status).
+	 */
 	public function __set( $key, $value ) {
 		$this->post->{ $key } = $value;
 	}
 
+	/**
+	 * Returning a WPCOM_JSON_API_Links instance if the post key is set to 'links', or the post key value.
+	 *
+	 * @param string $key The post key value.
+	 *
+	 * @return WPCOM_JSON_API_Links|string
+	 */
 	public function __get( $key ) {
-		if ( $key === 'links' ) {
-			require_once dirname( __FILE__ ) . '/class.json-api-links.php';
+		if ( 'links' === $key ) {
+			require_once __DIR__ . '/class.json-api-links.php';
 			return WPCOM_JSON_API_Links::getInstance();
 		}
 		return $this->post->{ $key };
 	}
 
+	/**
+	 * A function to either call a given function, or return an error if it doesn't exist.
+	 *
+	 * @param string $name A function name to be called.
+	 * @param mixed  $arguments Arguments to be passed into the given function.
+	 *
+	 * @return mixed|bool
+	 */
 	public function __call( $name, $arguments ) {
 		if ( is_callable( array( $this->post, $name ) ) ) {
 			return call_user_func_array( array( $this->post, $name ), $arguments );
 		} else {
-			trigger_error("Call to undefined method '{$name}'");
+			// phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_trigger_error
+			trigger_error(
+				esc_html(
+					sprintf(
+						/* translators: %s is the method name that has been called */
+						__( 'Call to undefined method %s', 'jetpack' ),
+						$name
+					)
+				)
+			);
 		}
 	}
 
-	public function __isset ( $name ) {
+	/**
+	 * Checking to see if a given property is set.
+	 *
+	 * @param string $name Property to check if set.
+	 *
+	 * @return bool
+	 */
+	public function __isset( $name ) {
 		return isset( $this->post->{ $name } );
 	}
 
+	/**
+	 * Defining a base get_like_count() function to be extended in the Jetpack_Post class.
```

Comment:
I'd skip the "Defining a base xyz function to be extended in the Jetpack_Post class" bit and just document what the function is supposed to do.

I'd also generally document that here, and in the Jetpack_Post class just do like
```php
/** @inheritDoc */
```
or
```php
/**
 * @inheritDoc
 *
 * Any relevant implementation notes, like "As this functionality isn't available locally, this implementation just returns 0."
 */
```